### PR TITLE
Add inv_ready event to signalize when the inventory can be used

### LINKED ISSUE #118: Add an event for when the inventory is loaded
That way we have an event to listen to to change slots and stuff

### PR DESCRIPTION
**Pull request by Gjum**
A new `inv_ready` event is emitted when the inventory got refreshed by the server, which happens after connection and when a new window is opened or closed.

Also sends `inv_open_window` when the player's inventory is opened initially.

The `inv_open_window` is delayed until all slots of the window are updated, so that a handler does not work with the wrong inventory.

Closes #118.

### CODE REVIEW

File: spock/plugins/helpers/inventory.py [hunk 1]
```diff
@@ -144,6 +145,16 @@ def __init__(self, ploader, settings):
         # stores the last click action for confirmation
         self.last_click = None
 
+        # to know when the inventory got synced with server
+        self.is_synchronized = False
```
Comment (barneygale) [inline spock/plugins/helpers/inventory.py]: Is there a situation where can this be defined? Should it be set to ``False`` in the class definition or initializer?

Comment (Gjum) [inline spock/plugins/helpers/inventory.py]: It should be an instance property, so it should be undefined when no instance (and thus no inventory to be synchronized) exists.

By the way, other plugins cannot access this anyways, as it is in `InventoryPlugin`, not in `InventoryBase`.

Comment (barneygale) [inline spock/plugins/helpers/inventory.py]: Ah sorry, I actually misread the diff and thought it wasn't being set in ``__init__``!

**Comment by nickelpro** [comment]
Only thing I'm unclear on @Gjum is the condition you're using to signal inventory synchronization. What are the significance of those id constants? I'm sure your behavior is correct I'd just like to understand before I merge

**Comment by Gjum** [comment]
@nickelpro When the server notices that the clients inventory state is wrong, it sends a window items, a set slot for the cursor, and a set slot for every non-empty slot. So by hooking into the cursor set slot, we know that at this point the clients inventory state is correct again.

**Comment by Gjum** [comment]
**TODO:** when merged, change all event names from `inv_` to `inventory_` (see also #139).

**Pull request merged**